### ISSUE TITLE
fix crash on exit when listening ports are already opened

### ISSUE THREAD

**Issue by astibal** [astibal/smithproxy#18]
```
root@xu-Standard-PC-Q35-ICH9-2009:~# 20-06-01 13:03:27.512181 <7f7628b06f40> Error    - [service]: Error initializing syslog server: 
20-06-01 13:03:27.512401 <7f7628b06f40> Fatal    - [service]: Error binding plain-tcp on port -130, exiting
20-06-01 13:03:27.513865 <7f7628b06f40> Fatal    - [service]: Error binding tls on port -130, exiting
20-06-01 13:03:27.514490 <7f7628b06f40> Fatal    - [service]: Error binding socks on port -130, exiting
20-06-01 13:03:27.514549 <7f7628b06f40> Fatal    - [service]: Error binding plain-rdr on port -130, exiting
20-06-01 13:03:27.514621 <7f7628b06f40> Fatal    - [service]: Error binding ssl-rdr on port -130, exiting
20-06-01 13:03:27.514704 <7f7628b06f40> Fatal    - [service]: Failed to setup proxies. Bailing!

[2]+  Segmentation fault      (core dumped) smithproxy.exe

```

**Comment by astibal** [comment]
forgot to mention it's a snap, not fully installed app (which still can suffer the same issue).

**Comment by astibal** [comment]
fixed by recent patch in 0.9.11
